Lib/test/test_os.py

```diff
@@ -2873,6 +2873,48 @@ def test_getfinalpathname_handles(self):
 
         self.assertEqual(0, handle_delta)
 
+    @support.requires_subprocess()
+    def test_stat_unlink_race(self):
+        # bpo-46785: the implementation of os.stat() falls back to reading
+        # the parent directory if CreateFileW() fails with a permission
+        # error. If reading the parent directory fails because the file or
+        # directory are subsequently unlinked, or because the volume or
+        # share are no longer available, then the original permission error
+        # should not be restored.
+        filename =  os_helper.TESTFN
+        self.addCleanup(os_helper.unlink, filename)
+        command = '''if 1:
```

Comment:
`textwrap.dedent()` is how we'd normally deal with indented string literals.

Reply:
Using `if 1:` is common in "test_os.py". I thought it was clever since the compiler optimizes it away, leaving no runtime cost.

Reply:
I don't have a strong opinion on this, however I'm leaning more towards `textwrap.dedent()`, as that's more explicit and readable to me. (Also, in Node.js for example, I would do this using the equivalent of `textwrap.dedent()` - [`strip-indent`](https://github.com/sindresorhus/strip-indent).)

Reply:
Done.